runtime/lua/vim/filetype.lua

```diff
@@ -273,6 +273,16 @@ local extension = {
   cfm = 'cf',
   cfi = 'cf',
   hgrc = 'cfg',
+  -- Extension match does not conflict with specific patterns such as '.*/etc/a2ps/.*%.cfg', etc.,
+  -- as it is done after those are tried to match
+  cfg = detect.cfg,
```

Comment:
can we generate/dispatch that? or add a separate "case insensitive" table?

Reply:
Both are possible, of course.

I'd say that generation will introduce avoidable complexity only for the sake of shaving off some lines of code.

Separate case insensitive matching sounds better and closer to what is done in Vim's 'filetype.vim'. There are two downsides:
- It won't fit all the cases ('php%d' and '%d' cases will still remain).
- It will add `string.lower()` call and extra table lookup when they can be avoided.

I think my personal preference would be to leave as is in this PR currently.

Reply:
> I'm just talking about the "full case-insensitive" extension -- my worry is that it's too easy to miss a case (beyond the cumbersomeness). A separate table for these 10 may be a worthwhile tradeoff.

I would agree If not for performance cost: the actual match after extra `string.lower()` and table lookup will is *extremely* rare. So this will essentially add extra avoidable cost for most cases.

And I certainly can see the "cumbersomeness". Especially if there will be something like "php%d%d" with its 100 matches :) But those cases are still can be handled via pattern: the cost will be a slight decrease in performance and (if merged) not working with future proposed `fast` flag.

> At the least, could you add this to the documentation for porting filetype patches? ("Prefer explicit lists over regex/patterns, especially for case insensitive matches".)

Sure, will do. Will adding to 'dev_vimpatch.txt' be enough or should I also add to the comment at the top of `pattern`?

Reply:
I think we should keep things in one place, so `:h dev_vimpatch`. It doesn't hurt to add a comment pointing to the help entry in a prominent place, too.

> the actual match after extra string.lower() and table lookup will is extremely rare

Well, there's also LPeg...

Reply:
Added to 'dev_vimpatch.txt' with slightly different wording ("explicit lists over regex/patterns" -> "explicit filenames/extensions over patterns".